java-checks/src/main/java/org/sonar/java/checks/LazyArgEvaluationCheck.java

```diff
@@ -288,6 +287,8 @@ private static class StringExpressionVisitor extends BaseTreeVisitor {
 
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
+      // assuming that tree.methodSymbol().returnType().isUnknown() is always false,
+      // because findStringArg() implicitly filters out unknown types due to symbolType() being null
       if (!isGetter(tree) && !isAnnotationMethod(tree)) {
```

Comment:
I removed the check we had in the previous PR:
```java
tree.methodSymbol().returnType().isUnknown() 
``` 
because it couldn't be covered by any test since unknown types generate a `MethodInvocationTree` where the arguments have a null `SymbolType`: those trees are filtered out by `findStringArg(MethodInvocationTree)`; the new logic never gets executed in such cases.
@alban-auzeill